builtin/providers/aws/resource_aws_alb_target_group_test.go

```diff
@@ -51,6 +51,127 @@ func TestAccAWSALBTargetGroup_basic(t *testing.T) {
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "aws_alb_target_group.test",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckAWSALBTargetGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSALBTargetGroupConfig_basic(targetGroupName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAWSALBTargetGroupExists("aws_alb_target_group.test", &before),
+				),
+			},
+			{
+				Config: testAccAWSALBTargetGroupConfig_updatedProtocol(targetGroupName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAWSALBTargetGroupExists("aws_alb_target_group.test", &after),
+					testAccChecALBTargetGroupRecreated(t, &before, &after),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSALBTargetGroup_changePortForceNew(t *testing.T) {
+	var before, after elbv2.TargetGroup
+	targetGroupName := fmt.Sprintf("test-target-group-%s", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "aws_alb_target_group.test",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckAWSALBTargetGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSALBTargetGroupConfig_basic(targetGroupName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAWSALBTargetGroupExists("aws_alb_target_group.test", &before),
+				),
+			},
+			{
+				Config: testAccAWSALBTargetGroupConfig_updatedPort(targetGroupName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAWSALBTargetGroupExists("aws_alb_target_group.test", &after),
+					testAccChecALBTargetGroupRecreated(t, &before, &after),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSALBTargetGroup_changeVpcForceNew(t *testing.T) {
+	var before, after elbv2.TargetGroup
+	targetGroupName := fmt.Sprintf("test-target-group-%s", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "aws_alb_target_group.test",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckAWSALBTargetGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSALBTargetGroupConfig_basic(targetGroupName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAWSALBTargetGroupExists("aws_alb_target_group.test", &before),
+				),
+			},
+			{
+				Config: testAccAWSALBTargetGroupConfig_updatedVpc(targetGroupName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAWSALBTargetGroupExists("aws_alb_target_group.test", &after),
+					testAccChecALBTargetGroupRecreated(t, &before, &after),
+				),
+			},
+		},
+	})
+}
+
+func testAccChecALBTargetGroupRecreated(t *testing.T,
+	before, after *elbv2.TargetGroup) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if before.TargetGroupArn == after.TargetGroupArn {
```

Comment:
This checks the pointers, not the values, so this will always pass given how we're construction the two group structs in `testAccCheckAWSALBTargetGroupExists`. If you compare the actual values, you find that only `TestAccAWSALBTargetGroup_changeNameForceNew` passes here because that's the only one that changes the name, thus changes the ARN (being a combo of region, account, and _name_). All the other tests fail when checking the values here, because the name hasn't changed, so the ARN hasn't changed.

I think it's sufficient to check that the new values were expecting were set. In this situation we don't actually care if it has a new ARN, or that the VPC changed in a `ForceNew` operation, we just care that if someone changed the VPC, that the change actually happens.